### ISSUE TITLE
Terminate task then restart: "Terminal with ID 2 does not exist"

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#31737]
I have a task with this config:

```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "npm",
            "script": "watch",
            "isBackground": true,
            "problemMatcher": "$gulp-tsc",
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```

I start the build task, then terminate it with `Tasks: Terminate task`. When I start it again I get an error popup, "Terminal with ID 2 does not exist".

**Comment by roblourens** [comment]
I can't start a task at all anymore until I reload the window, even if I create a new terminal.

**Comment by williscool** [comment]
getting this on windows while use `WSL`

with bash as the shell

**Comment by MagicRB** [comment]
This does still happen to me? Why closed?

**Comment by roblourens** [comment]
Ensure you're on the latest version, and if you see an issue like this, open a new issue with as many details as possible.
